From 571e601ab3bca15e2e652daa5b158fdaadebd799 Mon Sep 17 00:00:00 2001
Message-Id: <571e601ab3bca15e2e652daa5b158fdaadebd799.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Fri, 13 Jun 2014 08:25:37 +0530
Subject: [PATCH 099/312] REVERTME [VPG]: drm/i915: Handle primary flips only
 when primary enabled

Main plane interrupts are seen in VLV irq handler
even when Main plane is disabled. This patch adds a
check and handles the flips only when main plane (primary)
is enabled. This patch is a REVERTME. Proper fix will be
uploaded later.

Issue: GMIN-1455
Change-Id: I7133db08c5701d5634a82a389e5c421f828d59bf
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
Signed-off-by: Pallavi <pallavi.g@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index facef10..a6536da 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -1801,8 +1801,11 @@ static void valleyview_pipestat_irq_handler(struct drm_device *dev, u32 iir)
 			intel_pipe_handle_vblank(dev, pipe);
 
 		if (pipe_stats[pipe] & PLANE_FLIP_DONE_INT_STATUS_VLV) {
-			intel_prepare_page_flip(dev, pipe);
-			intel_finish_page_flip(dev, pipe);
+			/* Primary flips only when primary plane enabled */
+			if (I915_READ(DSPCNTR(pipe)) & DISPLAY_PLANE_ENABLE) {
+				intel_prepare_page_flip(dev, pipe);
+				intel_finish_page_flip(dev, pipe);
+			}
 		}
 		if (pipe_stats[pipe] & SPRITE0_FLIP_DONE_INT_STATUS_VLV) {
 				intel_prepare_sprite_page_flip(dev, pipe);
-- 
1.7.9.5

